### PR TITLE
Sumit Added LongestSubArrayHavingSumK code

### PR DESCRIPTION
**Pull request by sumitsantape30**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

This code helps to find the length of the longest sub-array having sum equal to the given value k.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

DWoC
Issue Addressed-  #1813

### CODE REVIEW

File: Java/other/LongestSubArrayHavingSumk.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+				if (maxLength < (i - map.get(sum - k)))
+					maxLength = i - map.get(sum - k);
+			}
+		}
+
+		return maxLength;
+	}
+
+	public static void main(String args[]) {
+		int[] arr = { 1, 2, 1, 0, 1, 1, 0 };
```
Comment (atarax665) [inline Java/other/LongestSubArrayHavingSumk.java:42]: Dynamic input is preferred.

File: Java/other/LongestSubArrayHavingSumk.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
```
Comment (atarax665) [inline Java/other/LongestSubArrayHavingSumk.java:1]: Add a beginner introduction of the program with the space and time complexity of your approach.

File: Java/other/LongestSubArrayHavingSumk.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	public static void main(String args[]) {
+		int[] arr = { 1, 2, 1, 0, 1, 1, 0 };
+		int n = arr.length;
+		int k = 4;
+		System.out.println("Length = " + lengthOfLongestSubarray(arr, n, k));
+	}
+}
+
+/*
```
Comment (atarax665) [inline Java/other/LongestSubArrayHavingSumk.java:49]: Add sample input/output here. Please refer to any recently merged Java PR to get an idea of the Java Code template that we follow.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Update the README file as well.

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
Update readme

**Comment by HarshCasper** [comment]
Closing the PR cause of no updates. Let me know if you would like this to be reopened.

**Comment by sumitsantape30** [comment]
Please Open this PR, sorry for the delays I had my exams so couldn't work on it.

**Pull request closed without merging**